sdk/identity/azure-identity/tests/test_username_password_credential.py

```diff
@@ -26,7 +26,7 @@ def test_no_scopes():
     """The credential should raise when get_token is called with no scopes"""
 
     credential = UsernamePasswordCredential("client-id", "username", "password")
-    with pytest.raises(ClientAuthenticationError):
+    with pytest.raises(ValueError):
```

Comment:
Why this happens?

I don't see we introduce breakings?

Reply:
This behavior, raising on `get_token()`, was added in 1.4.0b2, so changing the  exception type is within the bounds of preview releases.

As for why it changes here: this credential's `get_token` was previously decorated with `wrap_exceptions`, which re-raises all exceptions as `ClientAuthenticationError`. The decorator's purpose is to prevent leaking exceptions out of `msal`, but it also prevents raising built-in exceptions. This credential now inherits `get_token` from `InteractiveCredential`, which is not so decorated (but also prevents leaking `msal` exceptions). `ValueError` is a better exception for `get_token()` and I want all credentials to consistently raise it (`SharedTokenCacheCredential` is the only remaining credential which does not; #11553 lest we forget).

Reply:
I checked the history. Seems like in 1.3.1 (GA), we raised ClientAuthenticationError when scope it empty?

Reply:
Where do you see that? I don't find that behavior in 1.3.1's code: https://github.com/Azure/azure-sdk-for-python/blob/azure-identity_1.3.1/sdk/identity/azure-identity/azure/identity/_credentials/user.py#L145

Reply:
Get_token has wrap_exceptions decorated.

Reply:
In 1.3.1 `UsernamePasswordCredential.get_token` is decorated with `wrap_exceptions`, so it only raises `ClientAuthenticationError`. However, it doesn't necessarily raise at all when `get_token` is called with no scopes. I suspect it returns a token (surprise!) because the request is actually made by `msal`, which unintentionally uses default scopes in this case.

Reply:
Discussed offline.